### ISSUE TITLE
[BUG] - iOS Shortcut Not Working with Cloudflare Zero Tunnel Network Access

### ISSUE THREAD

**Issue by ChipSkylark37** [mealie-recipes/mealie#4742]
### First Check

- [X] This is not a feature request.
- [X] I added a very descriptive title to this issue (title field is above this).
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the Mealie documentation, with the integrated search.
- [X] I already read the docs and didn't find an answer.
- [ ] This issue can be replicated on the demo site (https://demo.mealie.io/).

### What is the issue you are experiencing?

I am using the latest Nightly build (as of 2024-12-18) and with a recently recreated PWA Shorcut.

I have read the documentation and trying to get either the iOS Shortcut (using Shortcuts App) or 'Send to PWA' working as others have discussed. Neither of which are working for me.

The iOS Shortcut does not see my instance and hangs at step one and I do not have the Share Sheet action of 'Sending to Mealie / PWA'. I did not test this with the Demo site - Is that possible to generate an API key to test with? 

### Steps to Reproduce

1) Setup shortcut via the following documentation:
https://docs.mealie.io/documentation/community-guide/ios/

OR 

1) Look for 'Share Sheet' in Safari to 'Send to PWA / Mealie' 

### Please provide relevant logs

I am unsure what logs would be required to troubleshoot this. 

### Mealie Version

Version
nightly

Build
613e4ef606908df885553a7c7f13e2c6da798d13 

### Deployment

Docker (Linux)

### Additional Deployment Details

The biggest caveat to this is that I use cloudflare Zero Tunnel Access and have it locked down with authorization where I need to log in first. I am thinking possible this could be the culprit, but don't know for certain as there's no logs to indicate. I am able to otherwise use Mealie inside of Safari without issue or through the PWA app that I have added through my instance.

**Comment by grzegor** [comment]
iOS shortcuts can't reach Mealie endpoints because they're not authenticated against your IdP from the access portal. You'd need to create an additional bypass rule on Cloudflare side with some form of a non-interactive method that can be used in shortcuts (the easiest being probably a bearer token in HTTP headers). This also has nothing to do with Mealie per se. You want two apps to talk to eachother (shortcuts and Mealie) but you've put a wall between them that you didn't account for in your flow.

**Comment by ChipSkylark37** [comment]
You’re totally right - I’ve been thinking about this and saw some other posts regarding the same. 

I’m going to try working on a way to get it to work but need to dig into the logs more. 

As of now, I’ll close this out.

**Comment by Borriborri** [comment]
Hi, Could you please verify? I made some adjustments to the CORS setting of the PWA. Delete and reinstall :)

**Comment by ChipSkylark37** [comment]
I entered my URL, Mealie API, and Gemini API and saved.

When trying to run, I get "Conversion Error. Get Dictionary Value failed because Shortcuts couldn't convert from Rich text to Dictionary."

This occurred under the step "Get (VALUE) for (slug) in (Contents of URL)"
